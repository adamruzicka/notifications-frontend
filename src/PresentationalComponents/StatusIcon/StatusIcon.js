import React, { Fragment } from 'react';
import {
    CheckCircleIcon,
    TimesCircleIcon,
    QuestionCircleIcon
} from '@patternfly/react-icons';
import PropTypes from 'prop-types';

export const StatusIcon = ({ status }) => {
    if (status && typeof(status) !== 'string') {
        return <Fragment>
            <CheckCircleIcon color='green' /> Success
        </Fragment>;
    } else if (status === 'unknown') {
        return <Fragment>
            <QuestionCircleIcon color='grey' /> Unknown
        </Fragment>;
    } else {
        return <Fragment>
            <TimesCircleIcon color='red' /> Error
        </Fragment>;
    }
};

StatusIcon.propTypes = {
    status: PropTypes.oneOfType([ PropTypes.bool, PropTypes.oneOf([ 'unknown' ]) ]).isRequired
};

export default StatusIcon;
