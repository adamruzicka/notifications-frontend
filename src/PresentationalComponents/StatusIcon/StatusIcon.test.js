import { StatusIcon } from './StatusIcon';

describe('StatusIcon', () => {
    it('expect to render CheckCircleIcon', () => {
        const wrapper = shallow(
            <StatusIcon status={ true } />
        );
        expect(wrapper.find('CheckCircleIcon').length).toBe(1);
        expect(wrapper.find('TimesCircleIcon').length).toBe(0);
        expect(wrapper.find('QuestionCircleIcon').length).toBe(0);
        expect(wrapper.render()).toMatchSnapshot();
    });
});

describe('StatusIcon', () => {
    it('expect to render TimesCircleIcon', () => {
        const wrapper = shallow(
            <StatusIcon status={ false } />
        );
        expect(wrapper.find('CheckCircleIcon').length).toBe(0);
        expect(wrapper.find('TimesCircleIcon').length).toBe(1);
        expect(wrapper.find('QuestionCircleIcon').length).toBe(0);
        expect(wrapper.render()).toMatchSnapshot();
    });
});

describe('StatusIcon', () => {
    it('expect to render QuestionCircleIcon', () => {
        const wrapper = shallow(
            <StatusIcon status={ 'unknown' } />
        );
        expect(wrapper.find('CheckCircleIcon').length).toBe(0);
        expect(wrapper.find('TimesCircleIcon').length).toBe(0);
        expect(wrapper.find('QuestionCircleIcon').length).toBe(1);
        expect(wrapper.render()).toMatchSnapshot();
    });
});
