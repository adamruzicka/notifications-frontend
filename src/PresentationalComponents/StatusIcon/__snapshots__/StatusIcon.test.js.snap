// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon expect to render CheckCircleIcon 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "fill": "green",
      "height": "1em",
      "role": "img",
      "style": "vertical-align:-0.125em",
      "viewBox": "0 0 512 512",
      "width": "1em",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "d": "M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z",
          "transform": "",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "d": undefined,
          "transform": undefined,
        },
        "x-attribsPrefix": Object {
          "d": undefined,
          "transform": undefined,
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": Object {
      "data": " Success",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "text",
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "data": " Success",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "text",
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "fill": undefined,
      "height": undefined,
      "role": undefined,
      "style": undefined,
      "viewBox": undefined,
      "width": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "fill": undefined,
      "height": undefined,
      "role": undefined,
      "style": undefined,
      "viewBox": undefined,
      "width": undefined,
    },
  },
  "1": Object {
    "data": " Success",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "aria-hidden": "true",
        "fill": "green",
        "height": "1em",
        "role": "img",
        "style": "vertical-align:-0.125em",
        "viewBox": "0 0 512 512",
        "width": "1em",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "d": "M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z",
            "transform": "",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "d": undefined,
            "transform": undefined,
          },
          "x-attribsPrefix": Object {
            "d": undefined,
            "transform": undefined,
          },
        },
      ],
      "name": "svg",
      "namespace": "http://www.w3.org/2000/svg",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "aria-hidden": undefined,
        "fill": undefined,
        "height": undefined,
        "role": undefined,
        "style": undefined,
        "viewBox": undefined,
        "width": undefined,
      },
      "x-attribsPrefix": Object {
        "aria-hidden": undefined,
        "fill": undefined,
        "height": undefined,
        "role": undefined,
        "style": undefined,
        "viewBox": undefined,
        "width": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "aria-hidden": "true",
            "fill": "green",
            "height": "1em",
            "role": "img",
            "style": "vertical-align:-0.125em",
            "viewBox": "0 0 512 512",
            "width": "1em",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z",
                "transform": "",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-hidden": undefined,
            "fill": undefined,
            "height": undefined,
            "role": undefined,
            "style": undefined,
            "viewBox": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-hidden": undefined,
            "fill": undefined,
            "height": undefined,
            "role": undefined,
            "style": undefined,
            "viewBox": undefined,
            "width": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "text",
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`StatusIcon expect to render QuestionCircleIcon 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "fill": "grey",
      "height": "1em",
      "role": "img",
      "style": "vertical-align:-0.125em",
      "viewBox": "0 0 512 512",
      "width": "1em",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "d": "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z",
          "transform": "",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "d": undefined,
          "transform": undefined,
        },
        "x-attribsPrefix": Object {
          "d": undefined,
          "transform": undefined,
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": Object {
      "data": " Unknown",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "text",
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "data": " Unknown",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "text",
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "fill": undefined,
      "height": undefined,
      "role": undefined,
      "style": undefined,
      "viewBox": undefined,
      "width": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "fill": undefined,
      "height": undefined,
      "role": undefined,
      "style": undefined,
      "viewBox": undefined,
      "width": undefined,
    },
  },
  "1": Object {
    "data": " Unknown",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "aria-hidden": "true",
        "fill": "grey",
        "height": "1em",
        "role": "img",
        "style": "vertical-align:-0.125em",
        "viewBox": "0 0 512 512",
        "width": "1em",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "d": "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z",
            "transform": "",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "d": undefined,
            "transform": undefined,
          },
          "x-attribsPrefix": Object {
            "d": undefined,
            "transform": undefined,
          },
        },
      ],
      "name": "svg",
      "namespace": "http://www.w3.org/2000/svg",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "aria-hidden": undefined,
        "fill": undefined,
        "height": undefined,
        "role": undefined,
        "style": undefined,
        "viewBox": undefined,
        "width": undefined,
      },
      "x-attribsPrefix": Object {
        "aria-hidden": undefined,
        "fill": undefined,
        "height": undefined,
        "role": undefined,
        "style": undefined,
        "viewBox": undefined,
        "width": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "aria-hidden": "true",
            "fill": "grey",
            "height": "1em",
            "role": "img",
            "style": "vertical-align:-0.125em",
            "viewBox": "0 0 512 512",
            "width": "1em",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z",
                "transform": "",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-hidden": undefined,
            "fill": undefined,
            "height": undefined,
            "role": undefined,
            "style": undefined,
            "viewBox": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-hidden": undefined,
            "fill": undefined,
            "height": undefined,
            "role": undefined,
            "style": undefined,
            "viewBox": undefined,
            "width": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "text",
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`StatusIcon expect to render TimesCircleIcon 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "fill": "red",
      "height": "1em",
      "role": "img",
      "style": "vertical-align:-0.125em",
      "viewBox": "0 0 512 512",
      "width": "1em",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "d": "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
          "transform": "",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "d": undefined,
          "transform": undefined,
        },
        "x-attribsPrefix": Object {
          "d": undefined,
          "transform": undefined,
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": Object {
      "data": " Error",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "text",
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "data": " Error",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "text",
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "fill": undefined,
      "height": undefined,
      "role": undefined,
      "style": undefined,
      "viewBox": undefined,
      "width": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "fill": undefined,
      "height": undefined,
      "role": undefined,
      "style": undefined,
      "viewBox": undefined,
      "width": undefined,
    },
  },
  "1": Object {
    "data": " Error",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "aria-hidden": "true",
        "fill": "red",
        "height": "1em",
        "role": "img",
        "style": "vertical-align:-0.125em",
        "viewBox": "0 0 512 512",
        "width": "1em",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "d": "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            "transform": "",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "d": undefined,
            "transform": undefined,
          },
          "x-attribsPrefix": Object {
            "d": undefined,
            "transform": undefined,
          },
        },
      ],
      "name": "svg",
      "namespace": "http://www.w3.org/2000/svg",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "aria-hidden": undefined,
        "fill": undefined,
        "height": undefined,
        "role": undefined,
        "style": undefined,
        "viewBox": undefined,
        "width": undefined,
      },
      "x-attribsPrefix": Object {
        "aria-hidden": undefined,
        "fill": undefined,
        "height": undefined,
        "role": undefined,
        "style": undefined,
        "viewBox": undefined,
        "width": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "aria-hidden": "true",
            "fill": "red",
            "height": "1em",
            "role": "img",
            "style": "vertical-align:-0.125em",
            "viewBox": "0 0 512 512",
            "width": "1em",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
                "transform": "",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "transform": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "transform": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-hidden": undefined,
            "fill": undefined,
            "height": undefined,
            "role": undefined,
            "style": undefined,
            "viewBox": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-hidden": undefined,
            "fill": undefined,
            "height": undefined,
            "role": undefined,
            "style": undefined,
            "viewBox": undefined,
            "width": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "text",
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
